### PR TITLE
Feature/pytest style for tests

### PR DESCRIPTION
**Pull request by alexis-thual**
Refactoring tests so that they use `fixtures` present in `./test/conftest.py`.

### CODE REVIEW

**Comment by alexis-thual** [comment]
In reference to #1835

**Comment by alexis-thual** [comment]
@robert-wagner I left a couple of questions open in this PR.
They can easily be found by looking for the word "TODO".
I usually leave a suggestion below.

File: test/torch/test_hook.py [hunk 1]
```diff
@@ -11,194 +11,190 @@
...
+    assert out[0] == m.some_params
+
+
+def test_torch_module_hook(workers):
+    """Tests sending and getting back torch nn module like nn.Linear"""
+    model = nn.Linear(2, 1)
+    model_ptr = model.send(workers["bob"])
+    res = model_ptr.get()
+
+    # TODO: shouldn't there be an assertion here?
```
Comment (LaRiffle) [inline test/torch/test_hook.py]: yes!

File: test/conftest.py [hunk 2]
```diff
@@ -21,9 +21,14 @@ def workers(hook):
     alice.add_workers([bob, james])
     james.add_workers([bob, alice])
     hook.local_worker.add_workers([alice, bob, james])
+    # TODO: should one set this boolean to true?
```
Comment (LaRiffle) [inline test/conftest.py]: usually it is already the case by default

File: test/torch/tensors/test_gc.py [hunk 3]
```diff
@@ -7,178 +7,163 @@
...
+    # delete pointer to pointer to tensor, which should automatically
+    # garbage collect the remote object on Bob's machine
+    del x_ptr_ptr
 
-    def test_explicit_garbage_collect_double_pointer(self):
-        """Tests whether deleting a pointer to a pointer garbage collects
-        the remote object too"""
+    # ensure bob's object was garbage collected
+    assert x.id not in workers["bob"]._objects
+    # TODO: shouldn't we check that alice's object was
```
Comment (LaRiffle) [inline test/torch/tensors/test_gc.py]: sure!

File: test/torch/tensors/test_gc.py [hunk 4]
```diff
@@ -7,178 +7,163 @@
...
+    # send tensor to bob and then pointer to alice
+    x = x.send(workers["bob"]).send(workers["alice"])
+    # ensure bob has tensor
+    assert x_id in workers["bob"]._objects
+    # delete pointer to pointer to tensor
+    del x
+    # ensure bob's object was garbage collected
+    assert x_id not in workers["bob"]._objects
+    # TODO: shouldn't we check that alice's object was
+    # garbage collected as well?
```
Comment (LaRiffle) [inline test/torch/tensors/test_gc.py]: Yup!

File: test/torch/tensors/test_gc.py [hunk 5]
```diff
@@ -7,178 +7,163 @@
...
+    # garbage collect the remote object on Bob's machine
+    x_ptr_ptr = "asdf"
 
-    def test_implicit_garbage_collect_double_pointer(self):
-        """Tests whether GCing a pointer to a pointer garbage collects
-        the remote object too"""
+    # ensure bob's object was garbage collected
+    assert x.id not in workers["bob"]._objects
+    # TODO: shouldn't we check that alice's object was
+    # garbage collected as well?
```
Comment (LaRiffle) [inline test/torch/tensors/test_gc.py]: same

File: test/torch/tensors/test_pointer.py [hunk 6]
```diff
@@ -77,54 +77,54 @@ def test_repeated_send(workers):
...
-    # get the gradient created from backpropagation manually
-    x_grad = workers["bob"]._objects[x.id_at_location].grad
-
-    # get the entire x tensor (should bring the grad too)
-    x = x.get()
-
-    # make sure that the grads match
-    assert (x.grad == x_grad).all()
+# def test_remote_autograd(workers):
+#     """Tests the ability to backpropagate gradients on a remote
```
Comment (LaRiffle) [inline test/torch/tensors/test_pointer.py]: why do we comment this?

Comment (alexis-thual) [reply test/torch/tensors/test_pointer.py]: It was already commented on `torch_1`, from which I branched.
We should probably try to uncomment it, but the test didn't pass so I left it as is.

Comment (LaRiffle) [reply test/torch/tensors/test_pointer.py]: Ah ok then go for it!

Comment (LaRiffle) [reply test/torch/tensors/test_pointer.py]: oh no actually it (test_remote_autograd)  has been un-commented now, see the merge conflict below

**Comment by iamtrask** [comment]
It'll take me a few days to get around to doing the merge conflict here (bit of a deadline for the rest of the base) - would be grateful for your help if you're willing.

**Pull request merged**